Description: Check if the systray is available before using it.
Author: Felix Geyer <debfx-pkg@fobos.de>

--- keepassx-0.4.3.orig/src/KpxConfig.h
+++ keepassx-0.4.3/src/KpxConfig.h
@@ -35,6 +35,8 @@
 #	define DEFAULT_MOUNT_DIR QString()
 #endif
 
+#include <QSystemTrayIcon>
+
 class KpxConfig{
 public:
 	friend class KpxBookmarks;
@@ -98,7 +100,11 @@ public:
 	bool lockOnInactivity(){return settings.value("Options/LockOnInactivity",false).toBool();}
 	int lockAfterSec(){return settings.value("Options/LockAfterSec",30).toInt();}
 	bool showStatusbar(){return settings.value("UI/ShowStatusbar",true).toBool();}
-	bool showSysTrayIcon(){return settings.value("Options/ShowSysTrayIcon",false).toBool();}
+	bool showSysTrayIcon(){
+		return QSystemTrayIcon::isSystemTrayAvailable() &&
+		qgetenv("XDG_CURRENT_DESKTOP") != "Unity" &&
+		settings.value("Options/ShowSysTrayIcon",false).toBool();
+	}
 	bool showToolbar(){return settings.value("UI/ShowToolbar",true).toBool();}
 	int toolbarIconSize(){return settings.value("UI/ToolbarIconSize",16).toInt();}
 	QString urlCmd(){return settings.value("Options/UrlCmd").toString();}
--- keepassx-0.4.3.orig/src/dialogs/SettingsDlg.cpp
+++ keepassx-0.4.3/src/dialogs/SettingsDlg.cpp
@@ -79,6 +79,10 @@ CSettingsDlg::CSettingsDlg(QWidget* pare
 	connect(this,SIGNAL(rejected()),SLOT(resetGlobalShortcut()));
 #endif
 	
+	if (!QSystemTrayIcon::isSystemTrayAvailable() || qgetenv("XDG_CURRENT_DESKTOP") == "Unity") {
+		CheckBox_ShowSysTrayIcon->setEnabled(false);
+	}
+	
 	listWidget->setCurrentRow(0);
 
 	//General (1)
